chainer/training/extensions/log_report.py

```diff
@@ -109,13 +121,36 @@ def __call__(self, trainer):
             # write to the log file
             if self._log_name is not None:
                 log_name = self._log_name.format(**stats_cpu)
-                with utils.tempdir(prefix=log_name, dir=trainer.out) as tempd:
-                    path = os.path.join(tempd, 'log.json')
-                    with open(path, 'w') as f:
-                        json.dump(self._log, f, indent=4)
 
-                    new_path = os.path.join(trainer.out, log_name)
-                    shutil.move(path, new_path)
+                if self._file_format == 'jsonl':
+                    # append the stats as a JSON object without CR/LF chars
+                    # and an LF as a delimiter
+                    stats_str = (json.dumps(stats_cpu, indent=None,
+                                            separators=(',', ':')) + '\n')
+                    self._log_str += stats_str.encode('utf-8')
+
+                    path = os.path.join(trainer.out, log_name)
+                    if path not in self._log_paths:
+                        # renew the log file (if exists) at first time
+                        # it also does when resuming (_log_paths will not be
+                        # serialized)
```

Comment:
How about `open(..., 'a')`?
Is it appropriate to overwrite when resuming?

Reply:
I'm, unfortunately, about to forget the details, but it seems it is to keep the behavior such that the reporter tries to resume but the log file is removed from the filesystem.